### DISCUSSION TITLE
Problem with using styled components with server-side rendering in next JS

### DISCUSSION THREAD

**Discussion by hdck007** [EddieHubCommunity/support#1428] [Q&A]
https://styled-components.com/docs/advanced#nextjs I have referred to this but it won't work for some pages.

**Comment by Rahma-AlWadhahi** [comment]
Does it show error messages in these pages? If so, what is it?

**Reply by hdck007** [reply]
That's the weirdest part no errors nothing. Also it works for single level routes like `host/admin` but not for `host/admin/settings`

**Reply by Ruchika30** [reply]
I have  a lame question , but is the style file being imported correctly , I mean you can chk the path

**Reply by hdck007** [reply]
If you are asking about importing CSS files then I am not using CSS here. I am using styled components. If you meant something else please correct me.

**Reply by Ruchika30** [reply]
_Got it . 

_"But there is only one problem: Server-side rendering for styled-components is not yet active. You can see this by the fact that the CSS styles are not served in the response of the server, in the HTML document."__

Did u check this blog 
https://javascript.plainenglish.io/server-side-rendering-styled-components-with-next-js-3197e527763b#:~:text=But%20there%20is%20only%20one,build%20version%20of%20our%20app.

**Reply by Ruchika30** [reply]
You can apply styled objects (css in js) , if you were using styled components for any other reason then thats  a different story

**Reply by hdck007** [reply]
https://github.com/vercel/next.js/tree/canary/examples/with-styled-components 
I have already gone through the process mentioned in the blog but not through the blog instead I checked nextjs official examples instead because it is the most up to date solution but it doesn't seems to work for two level routes in my case eg `host/admin/settings`
